src/ansys/units/quantity.py

```diff
@@ -5,8 +5,15 @@
 
 import ansys.units as ansunits
 
+try:
+    import numpy as np
 
-class Quantity(float):
+    _array = np
+except ImportError:
+    _array = None
+
+
+class Quantity:
```

Comment:
Now that we don't derive from `float` I think this means tests like `math.sqrt(q)` and some others now fail, you can see it in the pipeline.  I'm not certain right now how to get this working without the derivation.

Reply:
If we keep the `__float__` operator a quantity will work for this. Is there any downsides to having `__float__`?

Reply:
yeah, I'm on the fence, see the discussion above where I did a couple of experiments with pint and numpy.  I think it's ok to leave it out.

Reply:
@dnwillia-work The outcome of a discussion I had with @ansaboutin is that we would have the `__float__()` method. We will do the conversion to `float` if the quantity is dimensionless. The moot point is whether we also do it if the quantity is an angle. This would help to support interaction with trigonometric functions so we agreed to do that - that will return the value in radians (solid angle is also going to be supported). All other calls to convert to `float` would raise an exception.

Reply:
awesome, that sounds good.  Make sure to update the documentation about this behaviour.

Reply:
@ansaboutin please can you take care of the doc part also. Thanks!